pkg/controller/storagecluster/reconcile.go

```diff
@@ -1152,3 +1156,129 @@ func getMonCount(nodeCount int) int {
 
 	return count
 }
+
+//ensureJobTemplates returns spec of a job template
+func (r *ReconcileStorageCluster) ensureJobTemplates(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	osdCleanUpTemplate := &openshiftv1.Template{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sc.Name + "-osd-removal",
+			Namespace: sc.Namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, osdCleanUpTemplate, func() error {
+		osdCleanUpTemplate.Objects = []runtime.RawExtension{
+			{
+				Object: newCleanupJob(sc),
+			},
+		}
+		osdCleanUpTemplate.Parameters = []openshiftv1.Parameter{
+			{
+				Name:     "ROOK_IMAGE",
```

Comment:
We should be able to use the same image that is already being used for the rook operator. Can we lookup the rook operator image instead of requiring it as a parameter? The only input we should need from the admin is the osd id. The reconcile loop would also update the template whenever there is an upgrade to a new image.

Reply:
right can be done